bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2BridgeHandler.java

```diff
@@ -528,13 +529,15 @@ public void onResourcesEvent(List<Resource> resources) {
     }
 
     private void onResourcesEventTask(List<Resource> resources) {
-        logger.debug("onResourcesEventTask() resource count {}", resources.size());
+        int numberOfResources = resources.size();
+        logger.debug("onResourcesEventTask() resource count {}", numberOfResources);
+        Setters.mergeLightResources(resources);
+        if (numberOfResources != resources.size()) {
+            logger.debug("onResourcesEventTask() merged to {} resources", resources.size());
+        }
         getThing().getThings().forEach(thing -> {
-            ThingHandler handler = thing.getHandler();
-            if (handler instanceof Clip2ThingHandler) {
-                resources.forEach(resource -> {
-                    ((Clip2ThingHandler) handler).onResource(resource);
-                });
+            if (thing.getHandler() instanceof Clip2ThingHandler clip2ThingHandler) {
```

Comment:
Do you want to do all these conversions from Java 9 to Java 17 in a separate PR? (There are probably more than just this one..)

Reply:
I only changed it here because I was in the neighborhood. If there will be another PR for redoing those improvements, I will simply resolve this conflict.

I will not do general Java 17 improvements in context of this PR. i.e. those previously done in #15520 and [rejected by you](https://github.com/openhab/openhab-addons/pull/15520#issuecomment-1698194898).

Reply:
I don't mind if this change remains in the PR. Personally I have no plans to make another PR to convert Java 9 code styles to Java 17 styles, since Java 17 did not deprecate Java 9 styles, and so there is no need to change anything.